subsys/nrf_rpc/rpc_utils/server/dev_info.c

```diff
@@ -6,10 +6,205 @@
+{
+	struct coredump_cmd_copy_arg copy_params;
+
+	copy_params.offset = offset;
+	copy_params.buffer = (uint8_t *)buffer;
+	copy_params.length = length;
+
+	return coredump_cmd(COREDUMP_CMD_COPY_STORED_DUMP, &copy_params);
+}
+
+static int load_crash_info(struct nrf_rpc_crash_info *info)
+{
+	int rc;
+	int size;
+	uint16_t offset = 0;
+	uint16_t max_offset;
+	struct coredump_hdr_t dump_hdr;
+	struct coredump_arch_hdr_t arch_hdr;
+	struct arm_arch_block arch_blk;
+	struct coredump_threads_meta_hdr_t thread_meta_hdr;
+	struct coredump_mem_hdr_t mem_hdr;
+	struct dump_metadata metadata;
+	uint8_t header_id;
+
+	rc = coredump_cmd(COREDUMP_CMD_VERIFY_STORED_DUMP, NULL);
+	if (rc == 0) {
+		return -ENOENT;
+	} else if (rc != 1) {
+		return rc;
+	}
+
+	size = coredump_query(COREDUMP_QUERY_GET_STORED_DUMP_SIZE, NULL);
+	if (rc <= 0) {
+		return rc;
+	}
+
+	max_offset = offset + size;
+
+	rc = load_coredump_fragment(offset, &dump_hdr, sizeof(dump_hdr));
+	if (rc < 0) {
+		return rc;
+	}
+
+	info->reason = sys_le16_to_cpu(dump_hdr.reason);
+	offset = sizeof(dump_hdr);
+
+	while (offset < max_offset) {
+		rc = load_coredump_fragment(offset, &header_id, sizeof(uint8_t));
+		if (rc < 0) {
+			return rc;
+		}
+
+		switch (header_id) {
+		case COREDUMP_ARCH_HDR_ID:
+			/* Fill out available registers */
+			rc = load_coredump_fragment(offset, &arch_hdr, sizeof(arch_hdr));
+			if (rc < 0) {
+				return rc;
+			}
+
+			offset += sizeof(arch_hdr);
+
+			rc = load_coredump_fragment(offset, &arch_blk, arch_hdr.num_bytes);
+			if (rc < 0) {
+				return rc;
+			}
+
+			offset += arch_hdr.num_bytes;
+
+			info->pc = arch_blk.r.pc;
+			info->lr = arch_blk.r.lr;
+			info->xpsr = arch_blk.r.xpsr;
+			info->sp = arch_blk.r.sp;
+
+			break;
+		case THREADS_META_HDR_ID:
+			/* Intentionally skip over */
+			rc = load_coredump_fragment(offset, &thread_meta_hdr,
+						    sizeof(thread_meta_hdr));
+			if (rc < 0) {
+				return rc;
+			}
+
+			offset += sizeof(thread_meta_hdr) + thread_meta_hdr.num_bytes;
+
+			break;
+		case COREDUMP_MEM_HDR_ID:
+			rc = load_coredump_fragment(offset, &mem_hdr, sizeof(mem_hdr));
+			if (rc < 0) {
+				return rc;
+			}
+
+			size = mem_hdr.end - mem_hdr.start;
+			offset += sizeof(mem_hdr);
+
+			if (size >= sizeof(struct dump_metadata_header)) {
+				rc = load_coredump_fragment(offset, &metadata.header,
+							    sizeof(struct dump_metadata_header));
+				if (rc < 0) {
+					return rc;
+				}
+
+				offset += sizeof(struct dump_metadata_header);
+				size -= sizeof(struct dump_metadata_header);
+
+				if (memcmp((void *)metadata.header.magic, CRASH_INFO_MAGIC,
+					   sizeof(CRASH_INFO_MAGIC)) == 0) {
+					info->uuid = metadata.header.uuid;
+
+					if (metadata.header.has_assert_info) {
+						rc = load_coredump_fragment(offset,
+									    &metadata.assert_line,
+									    size);
+						if (rc < 0) {
+							return rc;
+						}
+
+						info->assert_line = metadata.assert_line;
```

Comment:
Looks like you should zero out `info` at the beginning of the function to make sure this is initialized.